interfaces/builtin/avahi_observe.go

```diff
@@ -19,14 +19,212 @@
 
 package builtin
 
+import (
+	"strings"
+
+	"github.com/snapcore/snapd/interfaces"
+	"github.com/snapcore/snapd/interfaces/apparmor"
+	"github.com/snapcore/snapd/interfaces/dbus"
+	"github.com/snapcore/snapd/release"
+)
+
 const avahiObserveSummary = `allows discovering local domains, hostnames and services`
 
 const avahiObserveBaseDeclarationSlots = `
   avahi-observe:
     allow-installation:
       slot-snap-type:
+        - app
         - core
     deny-auto-connection: true
+    deny-connection:
+      on-classic: false
+`
+
+const avahiObservePermanentSlotAppArmor = `
+network netlink,
+
+# Description: Allow operating as the avahi service. This gives
+# privileged access to the system.
+#include <abstractions/dbus-strict>
+
+dbus (send)
+    bus=system
+    path=/org/freedesktop/DBus
+    interface=org.freedesktop.DBus
+    member={Request,Release}Name
+    peer=(name=org.freedesktop.DBus, label=unconfined),
+
+dbus (receive, send)
+    bus=system
+    path=/org/freedesktop/DBus
+    interface=org.freedesktop.DBus
+    member=GetConnectionUnixProcessID
+    peer=(label=unconfined),
+
+dbus (receive, send)
+    bus=system
+    path=/org/freedesktop/DBus
+    interface=org.freedesktop.DBus
+    member=GetConnectionUnixUser
+    peer=(label=unconfined),
+
+# Allow binding the service to the requested connection name
+dbus (bind)
+    bus=system
+    name="org.freedesktop.Avahi",
+
+# Allow traffic to/from our path and interface with any method for unconfined
+# clients to talk to our service.
+dbus (receive, send)
+   bus=system
+   path=/org/freedesktop/Avahi{,/**}
+   interface=org.freedesktop.Avahi*
+   peer=(label=unconfined),
+
+# Allow traffic to/from org.freedesktop.DBus for Avahi service
+dbus (receive, send)
+   bus=system
+   path=/org/freedesktop/Avahi{,/**}
+   interface=org.freedesktop.DBus.*
+   peer=(label=unconfined),
+`
+
+const avahiObserveConnectedSlotAppArmor = `
```

Comment:
Also, since avahiObserveConnectedSlotAppArmor is used in avahi-control, let's mention that above avahiObserveConnectedSlotAppArmor:
```
# Note: avahiObserveConnectedSlotAppArmor is also used by avahi-control in AppArmorConnectedSlot
```